docs/DeveloperGuide.adoc

```diff
@@ -364,12 +474,320 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+**Cons:** May involve additional code for developers to consider. Users may filter unnecessary contacts that contain some of the keywords.
+
+---
+
+**Aspect:** Filter criteria: Match tolerance between the typed keywords and a person's tags +
+**Alternative 1 (current choice):** A contact will be filtered as long as any of its tag(s) matches any of the keyword(s) stated by the user. +
+**Pros:** User can widen his filter search on his list of contacts regardless of the combination of tags so the user may be able to find his contacts more easily.  +
+**Cons:** The filter command may include unnecessary filtered contacts when the user specifies more than one keyword and expects only the contacts with ALL such tags. Eg. A user may
+ indicate that he wants to filter contacts that has both `tutor` AND `professor` tags instead of contacts with either tutor` OR `professor` tags; the latter will result in more contacts
+  showing, some which may deem unnecessary to the user +
+**Alternative 2:** A contact will be filtered only if all its tags meet the required keyword(s). +
+**Pros:** The user is able to narrow down his filter to specific and more accurate contacts +
+**Cons:** The user might prefer a wider filter for either-or keywords as he might have forgotten which tag a particular contact was saved under. Narrowing down his filter may cause him some difficulties.
+
+---
+
+=== Sorting mechanism (Since v1.2)
+
+The sorting mechanism is achieved by the `SortCommand`. In order to sort the list of contacts, the default constructor of `SortCommand` creates an empty editable array list of contacts as shown below:
+
+    private ArrayList<ReadOnlyPerson> contactList;
+    public SortCommand() {
+        contactList = new ArrayList<>();
+    }
+
+The new list will take in a list of contacts from address book, sort them and overwrites the existing list of contacts in address book sorted by alphabetical order.
+
+The `SortCommand` inherits from `Command` instead of `UndoableCommands` as explained by the Undo/Redo mechanism earlier. There is no mechanism for sorting to be undone.
+
+The `SortCommand` is implemented in the following way:
+[source,java]
+----
+public class SortCommand extends Command {
+    @Override
+    public CommandResult execute() {
+        // ... sort logic ....
+    }
+}
+----
+When the command word `sort` or `st` is typed into UniCity, `AddressBookParser` will call `SortCommandParser`.
+
+Upon activating the application, an editable contactList containing ReadOnlyPerson will be empty at the beginning.
+
+When the user executes `SortCommand`, `sort` or `st`, to sort the list of contacts in address book. The current empty ArrayList of contactList will be sent to the model ........
+
+The following sequence diagram shows how the sort operation works:
+
+image::SortCommandSequenceDiagram.png[width="800"]
```

Comment:
This image has lower resolution in comparison to other diagrams. It might make reading difficult.